src/Search.php

```diff
@@ -4723,6 +4724,64 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
                     $name2 = 'firstname';
                 }
 
+                if (
+                    $searchopt[$ID]["datatype"] == 'dropdown'
+                    && is_a($itemtype, CommonITILObject::class, true)
+                    && isset($searchopt[$ID]["joinparams"]["beforejoin"]["joinparams"])
+                    && isset($searchopt[$ID]['joinparams']['beforejoin']['table'])
+                    && in_array($searchopt[$ID]['joinparams']['beforejoin']['table'], ['glpi_tickets_users', 'glpi_changes_users', 'glpi_problems_users'])
+                    && ((in_array($searchtype, ['equals', 'contains']) && $origin_nott) || (in_array($searchtype, ['notequals', 'notcontains']) && !$origin_nott))
+                ) {
+                   // could it be useful for other itemtypes and/or for other linked items (like glpi_groups_tickets)?
+                   // this block will be executed only when one negation is used (if two are used then it will not be executed)
+
+                    $link_table = $searchopt[$ID]['joinparams']['beforejoin']['table'];
+                    $orig_fk_field = getForeignKeyFieldForTable($orig_table);
+                    $inittable_fk_field = getForeignKeyFieldForTable($inittable);
+                    $addcondition = '';
+                    if (isset($searchopt[$ID]['joinparams']['beforejoin']['joinparams']) && isset($searchopt[$ID]['joinparams']['beforejoin']['joinparams']['condition'])) {
+                        $condition = $searchopt[$ID]['joinparams']['beforejoin']['joinparams']['condition'];
+                        if (is_array($condition)) {
+                            $it = new DBmysqlIterator(null);
+                            $condition = ' AND ' . $it->analyseCrit($condition);
+                        }
+
+                        $from         = ["`REFTABLE`", "REFTABLE", "`NEWTABLE`", "NEWTABLE"];
+                        $to           = ["`$orig_table`", "`$orig_table`", "`$link_table`", "`$link_table`"];
+                        $addcondition = str_replace($from, $to, $condition);
+                        $addcondition = $addcondition . " ";
+                    }
+
+                    $join = '';
+                    $crit = '';
+                    $notstr = 'NOT';
+                    $equalstr = '=';
+                   // crit for equals and notequals
+                    if (in_array($searchtype, ['equals', 'notequals'])) {
+                        if ($val == 0) {
+                           // $val == 0 special value to tell that there is no choice (= empty) in the dropdown
+                            $equalstr = 'IS NOT NULL';
```

Comment:
Should this be adapted to search type ?
Please add a test case for this special case.
```suggestion
                            $equalstr = $searchtype === 'equals' ? 'IS NULL' : 'IS NOT NULL';
```

Reply:
Hello @cedric-anne 
In fact these lines will be executed only and only when there is one and only one 'NOT', so this case does not need to be coded.
When there is no 'NOT' or two 'NOT' (= 'AND NOT' and 'NOT EQUALS') then the legacy code will be executed.
Thank you,
Regards,
Tomolimo